lib/inspec/utils/json_profile_summary.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Inspec
+  module Utils
+    #
+    # Inspec::Utils::JsonProfileSummary takes in certain information to identify a
+    # profile and then produces a JSON-formatted summary of that profile. It can
+    # return the results to STDOUT or a file. It is currently used in several
+    # places in the CLI such as `json`, `archive` and `artifact`.
+    #
+    #
+    module JsonProfileSummary
+      def self.produce_json(info:, write_path: "", suppress_output: false, logger:)
+        @logger = logger || Logger.new
```

Comment:
InSpec has a globally available object, `Inspec::Log`, which has been configured by the CLI options. You don't need to pass around a logger or create one of your own, nor keep a reference to it.
Just use it like this:
`Inspec::Log.info "Something happened"`

Reply:
Thanks for this! Very good to know and I've updated the repo :)